### DISCUSSION TITLE
Ory Use Case Question

### DISCUSSION THREAD

**Discussion by dm17** [ory/kratos#2520] [Q&A] [answered]
There are several sister websites that are related and we're about to add another... Their user bases are largely overlapping. I am thinking Ory might be a good fit to replace the auth and user management aspects of each site - to create a SSO solution.

1) Can you import hashed passwords into Ory in order to migrate each site without forcing users to reset their passwords?
2) Is there a standard way of merging identities during the import from multiple sites?
3) Can each site have a unique registration (keeping emails and/or usernames unique)?
4) Would Ory manage the registration process and allow for custom data fields in the registration form?

Does another auth & identity management solution seem like a better fit or does Ory cover the above?
Thanks!

**Comment by gen1us2k** [comment, marked as answer]
Hello, @dm17, Thanks for opening the discussion. 

>Can you import hashed passwords into Ory in order to migrate each site without forcing users to reset their passwords?

You can use [import identities guide](https://www.ory.sh/docs/guides/manage-identities/import-identity#importing-credentials) that supports importing BCrypt, PKBDF2, Argon2 Family Hashed Passwords.

> Is there a standard way of merging identities during the import from multiple sites?

AFAIK you can implement your own way of merging identities. That's because data models may vary from case to case. Someone uses a username as an identifier; someone uses an email or mobile number.

> Can each site have a unique registration (keeping emails and/or usernames unique)?

It depends. You can have one instance of Ory Kratos or Ory Cloud handle it for you, but it'll be a single-tenant instance. It means that all your sites will have the exact same identities, and one registered user will have access to all websites.
On the other hand, you can use multiple versions of Ory Kratos or Ory Cloud, for instance, one instance per website. In that case, you'll have isolated identities across websites. I think that multitenant setup will work for you.

> Would Ory manage the registration process and allow for custom data fields in the registration form?

You can check Identity [schema documentation](https://www.ory.sh/docs/kratos/concepts/identity-schema) to customize data in the registration form.

**Reply by vinckr** [reply]
Hey @dm17 ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
